### PR TITLE
CC-753 upgrade tester utils

### PR DESCRIPTION
**Pull request by libmartinito**

### CODE REVIEW

File: internal/stages_test.go [hunk 1]
```diff
@@ -1,157 +1,53 @@
...
+			ExpectedExitCode: 0,
+			StdoutFixturePath: "./test_helpers/fixtures/write_tree",
+			NormalizeOutputFunc: normalizeTesterOutput,
+		},
+	}
+
+	tester_utils_testing.TestTesterOutput(t, testerDefinition, testCases)
+}
+
+func normalizeTesterOutput(testerOutput []byte) []byte {
```
Comment (libmartinito) [inline internal/stages_test.go:51]: it seems that the `CompareOutputWithFixture` function in tester utils requires a `NormalizeOutputFunc`. is it better that this is always required? not sure how to normalize output for git, so i just returned the output

**Comment by libmartinito** [comment]
there aren't tests for some of the stages. should i make a linear test for that for each tester or no

File: internal/stages_test.go (hunk 1 above)
Comment (rohitpaulk) [reply internal/stages_test.go:51]: Yep, it's okay to just return the output if that works - some challenges will have randomness involved (like with TCP port numbers), that's where the normalization function is useful.

**Comment by rohitpaulk** [comment]
> there aren't tests for some of the stages. should i make a linear test for that for each tester or no

Nope, it's fine - we'd want to retain whatever tests we have, don't need to add new ones as part of this migration. 

> appreciate if you could take a look at one instance before doing the other testers

@libmartinito you've got failing tests - other than that looks good. Does `make test_with_git` work locally?

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
(comment added)

**Comment by libmartinito** [comment]
yep it works locally. the problem was the randomness in the file names, contents, hashed values. updated to be more deterministic.

**Comment by rohitpaulk** [comment]
@libmartinito the randomness in tests is intentional. We do this to prevent users from accidentally passing the stage by hardcoding values. Let's find another way around it - check what other challenges do

**Pull request merged**